### ISSUE TITLE
NullPointerException in cmp gateway using xi:cmp-update-p12 because no profile is set.

### ISSUE THREAD

**Issue by MartenLeckebusch-DB** [xipki/xipki#306]
I am using xipki in the v6.4.0 version, running the cmp gateway and ca server in apache tomcat.
Trying to update a certificate using xi:cmp-update-p12 in the karaf xipki cli results in an exception in the cmp gateway because no certprofile is set. 
This seems to be because while xi:cmp-enroll-p12 requires the --profile option, the xi:cmp-update-p12 does not have such an option nor does the gateway set it anyhow.
The log of the cmp gateway reveals the following exception, which is raised in the CmpResponder.enrollCerts() in line 361: 

```
java.lang.NullPointerException: value may not be null
	at java.util.Objects.requireNonNull(Objects.java:235) ~[?:?]
	at org.xipki.util.Args.notNull(Args.java:114) ~[util-6.3.1.jar:?]
	at org.xipki.audit.AuditEventData.<init>(AuditEventData.java:23) ~[audit-6.3.1.jar:?]
	at org.xipki.audit.AuditEvent.addEventData(AuditEvent.java:122) ~[audit-6.3.1.jar:?]
	at org.xipki.ca.gateway.cmp.CmpResponder.enrollCerts(CmpResponder.java:361) ~[cmp-gateway-6.4.0.jar:?]
        ...
```

I have not tried this with newer xipki versions, but I would assume that the error persists.

Kind Regards!

**Comment by MartenLeckebusch-DB** [comment]
Depends on when you're planning to do a new release. 
I have catched a cold so I'll probably not work on it again before christmas :)

**Comment by xipki** [comment]
Since this feature (key-update) is not used widely, I am not planing to release xipki only due to this bug. But if you wish, I can do it.

**Comment by xipki** [comment]
Released v6.5.3.
